homeassistant/components/enphase_envoy/binary_sensor.py

```diff
@@ -86,31 +125,60 @@ async def async_setup_entry(
             for encharge in envoy_data.encharge_inventory
         )
 
+    if envoy_data.enpower:
+        entities.extend(
+            EnvoyEnpowerBinarySensorEntity(coordinator, description)
+            for description in ENPOWER_SENSORS
+        )
+
+    if envoy_data.dry_contact_status:
+        entities.extend(
+            EnvoyRelayBinarySensorEntity(coordinator, relay)
+            for relay in envoy_data.dry_contact_status
+        )
     async_add_entities(entities)
 
 
-class EnvoyEnchargeBinarySensorEntity(
+class EnvoyBaseBinarySensorEntity(
     CoordinatorEntity[EnphaseUpdateCoordinator], BinarySensorEntity
 ):
```

Comment:
The body of this function is the same as `sensor.py` so we could make an `entity.py` and make a 

`class EnvoyBaseEntity(CoordinatorEntity[EnphaseUpdateCoordinator])` and share it between `sensor.py` and `binary_sensor.py` (change description to `description: EntityDescription`)

Than you could have in `binary_sensor.py`

`class EnvoyBaseBinarySensorEntity(EnvoyBaseEntity, BinarySensorEntity)`

Than you could have in `sensor.py`

`class EnvoyBaseSensorEntity(EnvoyBaseEntity, SensorEntity)`

Reply:
Might be better to do that in another followup PR